.github/workflows/build-binary.yml

```
@@ -0,0 +1,122 @@
+      - id: determine-version
+        shell: pwsh
+        run: |
+          if ( ${{github.event_name}} -eq 'release') {
+            echo "version=${{ github.event.release.tag_name}}" >> $GITHUB_OUTPUT
+          } else if (${{inputs.sem-version}}) {
+             echo "version=${{ inputs.sem-version}}" >> $GITHUB_OUTPUT
+          }
+          Write-Error "Version neither via input nor by tag specified. Aborting"
+          exit 1
+      - id: determine-number
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2006:style:1:13: Use $(...) notation instead of legacy backticks `...`

(shellcheck)